[
  {
    "project": "tiangolo/fastapi",
    "commit": "3ed112e8a93d1b6fdcdf3f9fe9e8df986114cc1c",
    "filename": "fastapi/exceptions.py",
    "filename_after_commit": "/home/chowyi/TypeAnnotation_Study/GitHub/tiangolo-fastapi/fastapi/exceptions.py",
    "file_hunks_size": 2,
    "min_patch_found": false,
    "single_hunk": false,
    "fit_TFix": false,
    "delete_only_patch": false,
    "has_suppression_all_hunks": false,
    "full_warning_msg": "fastapi/exceptions.py:7:32 Incompatible variable type [9]: detail is declared to have type `str` but is used as type `None`.",
    "message": " detail is declared to have type `str` but is used as type `None`.",
    "rule_id": "Incompatible variable type [9]",
    "warning_line_no": 7,
    "warning_line": "        self, status_code: int, detail: str = None, headers: dict = None"
  }
]